Sprint-1/interpret/to-pounds.js

```diff
@@ -25,3 +25,27 @@ console.log(`£${pounds}.${pence}`);
 
 // To begin, we can start with
 // 1. const penceString = "399p": initialises a string variable with the value "399p"
+
+/* 1. Set the string with pence:
+   1.1. Let `penceString` = "399p" (string representing the number of pence)
+
+2. Remove the trailing "p" character from the string:
+   2.1. Get the substring of `penceString` excluding 
+   the last character (the "p"), this will be `penceStringWithoutTrailingP`
+
+3. Pad the pence number string with leading zeros to make it 3 characters long:
+   3.1. Use the `padStart(3, "0")` method on `penceStringWithoutTrailingP` 
+   to add leading zeros, saving the result in `paddedPenceNumberString`
+
+4. Extract the number of pounds:
+   4.1. Get the substring of the first 1-2 digits from 
+   `paddedPenceNumberString`, which will give the number 
+   of pounds, saved as `pounds`
+
+5. Extract the number of pence:
```

Comment:
Can we expect the code to work as intended even if we deleted `.padEnd(2, "0")` from the code?

Reply:
Yes, I think that if you remove this part of the code, everything will work as intended.
When I was designing the algorithm for this script, I noticed this and thought "why?". After all, we first make a string of length 3 in case there is one significant penny. There are no separators, therefore we will not have a situation where we need to, for example, convert 2 to 20. If we had, for example, 3.2 at the input, where 3 is pounds and 2 is 20 pence, this code would be simply necessary. But here, this is just an extra load on the processor, system interfaces and data buses. In the context of our course, this is **dead code**.